engine/src/main/scala/cromwell/engine/backend/CanUseGcsFilesystem.scala

```diff
@@ -0,0 +1,44 @@
+package cromwell.engine.backend
+
+import java.nio.file.FileSystem
+
+import cromwell.core.WorkflowOptions
+import cromwell.filesystems.gcs.{GcsFileSystemProvider, GoogleAuthMode, GoogleConfiguration}
+import lenthall.config.ScalaConfig._
+import EnhancedWorkflowOptions._
+
+import scalaz.NonEmptyList
+
+trait CanUseGcsFilesystem { self: Backend =>
+
+  protected def gcsFilesystem(options: WorkflowOptions): Option[FileSystem] = {
+
+    def gcsFilesystemMustHaveAuth: Option[String] = {
+      backendConfig.getConfigOption("filesystems.gcs") map {
+        _.getStringOr("auth", throw new RuntimeException(s"No auth in filesystems.gcs for backend '$name'.")) }
+    }
+
+    def gcsFilesystemAuthMustBeLegit(authName: String): GoogleAuthMode = {
+      def prependName(errs: NonEmptyList[String]): String = s"For backend '$name': " + errs.list.mkString("\n")
```

Comment:
Is it possible to use https://github.com/broadinstitute/lenthall/blob/develop/src/main/scala/lenthall/exception/ExceptionAggregation.scala#L7 ?
It preserves the list of errors separate form the message so we can report / display them more easily